hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/AzureBlobFileSystemStore.java

```diff
@@ -1314,7 +1352,30 @@ private String convertXmsPropertiesToCommaSeparatedString(final Hashtable<String
     return properties;
   }
 
+  private boolean isKeyForPrefixSet(String key, Set<String> dirSet) {
+
+    for (String dir : dirSet) {
+      if (dir.isEmpty() || key.startsWith(dir)) {
+        return true;
+      }
+
+      try {
+        URI uri = new URI(dir);
+        if (null == uri.getAuthority()) {
+          if (key.startsWith(dir + "/")){
+            return true;
+          }
+        }
+      } catch (URISyntaxException e) {
+        LOG.info("URI syntax error creating URI for {}", dir);
```

Comment:
How about throwing an AzureBlobFileSystemException from here. So that the customer will get to know that the config is not correct.

Reply:
this is used for every file being created, returning true or false. Raising an exception can be a problem.

Reply:
but what if only one comma separated value is incorrect, we would be throwing an exception adn crashing the app/jvm?